File: zig/test/syscalls/args_print.c
#include <stdio.h>
#include <time.h>
#include <sys/stat.h>
#include <stdlib.h>


void print_space(int repeat) {
    for(int i = 0; i < repeat; i++) {
        printf("%d - Hello world ", i);
    }
}

void print_new_line(int repeat) {
    for(int i = 0; i < repeat; i++) {
        printf("%d - Hello world\n", i);
    }
}




int main(int argc, char **argv) {
    if (argc != 2) {
        printf("Usage <integer>");
        return 1;
    }

    int repeat = atoi(argv[1]);


    clock_t time_space = clock();
    print_space(repeat);
    time_space = clock() - time_space;
    double time_space_taken = (double)time_space / CLOCKS_PER_SEC;

    clock_t time_new_line = clock();
    print_new_line(repeat);
    time_new_line = clock() - time_new_line;
    double time_new_line_taken = (double)time_new_line / CLOCKS_PER_SEC;


    printf("Sapce time: \t%lf\nNew Line time: \t%lf\n", time_space_taken, time_new_line_taken);

    return 0;
}

File: zig/test/syscalls/args_print.zig
const c = @cImport({
    @cInclude("stdio.h");
    @cInclude("stdlib.h");
    @cInclude("time.h");
    @cInclude("sys/stat.h");
});

fn print_space(repeats: c_int) void {
    for (0..@intCast(repeats)) |i| {
        _ = c.printf("%lu - Hello world ", i);
    }
}

fn print_new_line(repeats: c_int) void {
    for (0..@intCast(repeats)) |i| {
        _ = c.printf("%lu - Hello world\n", i);
    }
}

pub export fn main(argc: c_int, argv: [*]const [*:0]const u8) c_int {
    if (argc != 2) {
        _ = c.printf("Usage: %s <integer>\n", argv[0]);
        return 1;
    }

    const repeats: c_int = c.atoi(argv[1]);

    var space_time: c.clock_t = c.clock();
    print_space(repeats);
    space_time = c.clock() - space_time;
    const space_time_taken: f64 = @as(f64, @floatFromInt(space_time)) / c.CLOCKS_PER_SEC;

    var new_line_time: c.clock_t = c.clock() - space_time;
    print_new_line(repeats);
    new_line_time = c.clock() - new_line_time;
    const new_line_time_taken: f64 = @as(f64, @floatFromInt(new_line_time)) / c.CLOCKS_PER_SEC;
    _ = c.printf("\nSpace time: \t%lf sec \nNew Line time: \t%lf sec\n", space_time_taken, new_line_time_taken);

    _ = c.fflush(c.stdout);
    return 0;
}
